bot/extensions/reminder_cog.py

```diff
@@ -0,0 +1,128 @@
+import re
+from datetime import datetime, timedelta
+from discord import Embed
+from discord.ext.commands import Cog, hybrid_command
+from pytz import timezone
+from typing import Match
+
+
+class ReminderCog(
+    Cog,
+    name="Reminder",
+    description="Handles reminder functionalities within the Discord bot.",
+):
+    """A Discord Cog that manages user reminders."""
+
+    def __init__(self, bot):
+        self.bot = bot
+        self.jobs = []
+        self.timezone = timezone("US/Eastern")
+
+    def cog_unload(self):
+        """Clean up any jobs this cog created."""
+        for job in self.jobs:
+            self.bot.scheduler.remove_job(job.id)
+
+    def _build_embed(self, title: str, message: str, author: str) -> Embed:
+        """Builds a Discord embed with the given description.
+
+        :param title: The title of the embed.
+        :param message: The description/message of the embed.
+        :param author: The author of the embed.
+
+        :return: The constructed Embed object.
+        """
+        embed = Embed(
+            color=self.bot.default_color,
+            title=f"**{title}**",
+            description=message,
+            timestamp=datetime.now(),
+        )
+        embed.set_author(name=author)
+        embed.set_thumbnail(
+            url=(
+                "https://external-content.duckduckgo.com/iu/?u="
```

Comment:
I am worried to use an url like this and that at some point it stop working because it's invalid anymore, but we don't really have an asset folder. 

We could add one, but for now it should be fine.